### PR TITLE
Add comdb2_index_usage tables

### PR DESCRIPTION
**Pull request by mponomar**
```
$ cdb2sql --tabular mikedb 'select * from comdb2_index_usage'
+------------+--------+----------+-----------------+-------+---------------+
| table_name | ix_num | csc_name | sql_name        | steps | non_sql_steps |
+------------+--------+----------+-----------------+-------+---------------+
| 't'        | 0      | 'T_A'    | '$T_A_347F9EB9' | 0     | 0             |
| 't'        | 1      | 'T_B'    | '$T_B_AD76CF03' | 0     | 0             |
+------------+--------+----------+-----------------+-------+---------------+
$ cdb2sql mikedb 'select * from t where a=1' >/dev/null
$ cdb2sql mikedb 'select * from t where a=1' >/dev/null
$ cdb2sql mikedb 'select * from t where a=1' >/dev/null
$ cdb2sql mikedb 'select * from t where b=1' >/dev/null
$ cdb2sql --tabular mikedb 'select * from comdb2_index_usage'
+------------+--------+----------+-----------------+-------+---------------+
| table_name | ix_num | csc_name | sql_name        | steps | non_sql_steps |
+------------+--------+----------+-----------------+-------+---------------+
| 't'        | 0      | 'T_A'    | '$T_A_347F9EB9' | 404   | 0             |
| 't'        | 1      | 'T_B'    | '$T_B_AD76CF03' | 101   | 0             |
+------------+--------+----------+-----------------+-------+---------------+
$



```

### CODE REVIEW

File: sqlite/ext/comdb2/indexuse.c [hunk 1]
```diff
@@ -0,0 +1,110 @@
+/*
+   Copyright 2017 Bloomberg Finance L.P.
```
Comment (nirbhayc) [inline sqlite/ext/comdb2/indexuse.c]: 2020

File: sqlite/ext/comdb2/indexuse.c [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        if (strncmp(db->tablename, "sqlite_stat", strlen("sqlite_stat")) == 0)
+            continue;
+        logmsg(LOGMSG_USER, "table '%s'\n", db->tablename);
+        for (int ixnum = 0; ixnum < db->nix; ixnum++) {
+            if (nix == allocated) {
+                allocated = allocated * 2 + 16;
+                struct index_usage *n = realloc(ixs, allocated * sizeof(struct index_usage));
+                if (n == NULL) {
+                    free_index_usage(ixs, nix);
+                    unlock_schema_lk();
```
Comment (nirbhayc) [inline sqlite/ext/comdb2/indexuse.c]: return 1; ?

File: sqlite/ext/comdb2/indexuse.c [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    return 0;
+}
+
+int systblSQLIndexStatsInit(sqlite3 *db) {
+    int rc = create_system_table(db, "comdb2_index_usage", &systblIndexUsageModule,
+            get_index_usage, free_index_usage, sizeof(index_usage),
+            CDB2_CSTRING, "table", -1, offsetof(index_usage, tablename),
+            CDB2_INTEGER, "ixnum", -1, offsetof(index_usage, ixnum),
+            CDB2_CSTRING, "cscname", -1, offsetof(index_usage, cscname),
+            CDB2_CSTRING, "sqlname", -1, offsetof(index_usage, sqlname),
```
Comment (nirbhayc) [inline sqlite/ext/comdb2/indexuse.c]: I'd snake_case above 3 column names.

**Comment by mponomar** [comment]
Also needs a test.

**Pull request merged**